scheduler/util.go

```diff
@@ -376,11 +377,18 @@ func taintedNodes(state State, allocs []*structs.Allocation) (map[string]*struct
 	return out, nil
 }
 
-// shuffleNodes randomizes the slice order with the Fisher-Yates algorithm
-func shuffleNodes(nodes []*structs.Node) {
+// shuffleNodes randomizes the slice order with the Fisher-Yates
+// algorithm. We seed the random source with the eval ID (which is
+// random) to aid in postmortem debuggging of specific evaluations and
+// state snapshots.
+func shuffleNodes(evalID string, nodes []*structs.Node) {
+
+	seed, _ := binary.Varint([]byte(evalID))
```

Comment:
I don't think this is technically correct

> Varint decodes an int64 from buf and returns that value and the number of bytes read (> 0).

I think we only get 1 byte of randomness: 
https://go.dev/play/p/OttVV5vI_Xu

Though TBH that's probably fine ~

Reply:
Huh! I knew that the 2nd return value was the bytes read, but I didn't realize it'd only read one byte. But now I'm trying to figure out what circumstance it could ever return a negative number for overflow. The docs for that method are weird.


In any case, I also just realized that if we were to do https://github.com/hashicorp/nomad/issues/11779 this would result in all the evals effectively having the same seed for a long stretch of time. I'll update this to pull the bytes off the _end_ of the eval ID (which are random even if we do #11779) and use the `binary.BigEndian.Uint32` functions which appear to read as many bytes as you ask for.